pkgs/development/compilers/z88dk/Makefile.patch

```diff
@@ -0,0 +1,60 @@
+--- z88dk.orig/Makefile	1970-01-01 01:00:01.000000000 +0100
++++ z88dk/Makefile	2018-02-16 13:28:44.979280387 +0100
+@@ -8,32 +8,32 @@
+ # ---> Configurable parameters are below his point
+ prefix ?= /usr/local
+ prefix_share = $(prefix)/share/z88dk
+-git_rev = $(shell git rev-parse --short HEAD)
+-git_count = $(shell git rev-list --count HEAD)
+-version := $(shell date +%Y%m%d)
++git_rev ?= $(shell git rev-parse --short HEAD)
++git_count ?= $(shell git rev-list --count HEAD)
++version ?= $(shell date +%Y%m%d)
+ 
+ INSTALL ?= install
+ CFLAGS ?= -O2
+ CC ?= gcc
+ # Prefix for executables (eg z88dk-, hence z88dk-z80asm, z88dk-copt etc)
+-EXEC_PREFIX ?= 
++EXEC_PREFIX ?=
+ CROSS ?= 0
+ 
+ # --> End of Configurable Options
+ 
+-export CC INSTALL CFLAGS EXEC_PREFIX CROSS 
++export CC INSTALL CFLAGS EXEC_PREFIX CROSS
+ 
+ all: setup appmake copt zcpp ucpp sccz80 z80asm zcc zpragma zx7 z80nm lstmanip ticks z80svg testsuite z88dk-lib
+ 
+ setup:
+ 	$(shell if [ "${git_count}" != "" ]; then \
+-	    echo '#define PREFIX "${prefix_share}"' > src/config.h; \
++	    echo '#define PREFIX "${DESTDIR}${prefix_share}"' > src/config.h; \
+ 	    echo '#define UNIX 1' >> src/config.h; \
+ 	    echo '#define EXEC_PREFIX "${EXEC_PREFIX}"' >> src/config.h; \
+ 	    echo '#define Z88DK_VERSION "${git_count}-${git_rev}-${version}"' >> src/config.h; \
+ 	fi)
+ 	$(shell if [ ! -f src/config.h ]; then \
+-	    echo '#define PREFIX "${prefix_share}"' > src/config.h; \
++	    echo '#define PREFIX "${DESTDIR}${prefix_share}"' > src/config.h; \
+ 	    echo '#define UNIX 1' >> src/config.h; \
+ 	    echo '#define EXEC_PREFIX "${EXEC_PREFIX}"' >> src/config.h; \
+ 	    echo '#define Z88DK_VERSION "unknown-unknown-${version}"' >> src/config.h; \
+@@ -103,7 +103,7 @@
+ 	cd libsrc ; $(MAKE) install
+ 
+ install: install-clean
+-	install -d $(DESTDIR)/$(prefix) $(DESTDIR)/$(prefix_share)/lib 
++	install -d $(DESTDIR)/$(prefix) $(DESTDIR)/$(prefix_share)/lib
```

Comment:
What is the functional difference here?

Reply:
nothing, something remove  trailing whitespaces and my diff is a bit bigger than it could. But btw, i'd prefer fix that upstream later.

Reply:
It might be helpful to refer to a pull request, if it exists - Then future contributor have an easier time to update `z88dk`, if changes were applied upstream.

Reply:
sure, i did it for squashfuse, but i wont do that for z88dk for the moment (i can't contribute all project with fixes, and i need to use more z88dk to do so). I think "Release early, release often" was a good principle . I spent two day on this , i think future contributor would prefer modify existing not perfect derivation than start from scratch. Btw i've 6 open PR for nix, so i see the quality required is very high for new derivation.

Reply:
Your open PR requests are more likely related to backlog nixpkgs maintainer have due lack of time.

Reply:
ok. As you can see, i commit upstream project i do nix derivation, but sometime, i don't want to waste time to argue , and this patch was some of waste of time. They modify this makefile very often, since z88dk team are in a newlib , new compatibility with sdcc and so, that this was just a not so quick fix to let it works in nix and make stuff with.

Reply:
they were pretty quick merging the change btw: https://github.com/z88dk/z88dk/pull/612

Reply:
i've to admit. and i was quick to merge the fixing on #fc094db . I've other things to fix as examples optional installation and others stuff. But i'm happy to see nix curators are alive, i've so much to do in it.